server/server.go

```diff
@@ -42,72 +55,362 @@ func HandleWebsocketConnection(serverHTTPResponse http.ResponseWriter, clientHTT
 	if websocketConnectionErr != nil {
 		// If there was an error while upgrading the HTTP connection to a WebSocket connection return a relevant error
 		http.Error(serverHTTPResponse, "Unable to upgrade HTTP connection to a WebSocket connection", 400)
+		// Close the websocket connection
+		websocketConnection.Close()
 		return
 	}
-
 	// Require successful authentication before allowing a client to send a message
-	_, websocketConnectionStateErr := requireAuth(websocketConnection)
-	if websocketConnectionStateErr != nil {
+	authenticatedIdentityPublicKeyHexClient, websocketConnectionrequestAuthErr := requestAuth(websocketConnection)
```

Comment:
Little typo `websocketConnectionrequestAuthErr` -> `websocketConnectionRequestAuthErr`

Reply:
Will address it so that naming is more consistent.